hunting/okta/docs/queries/defense_evasion_failed_oauth_access_token_retrieval_via_public_client_app.toml

```diff
@@ -0,0 +1,40 @@
+[hunt]
+author = "Elastic"
+description = """
+This hunting query identifies when a public client app fails to retrieve an OAuth access token using client credentials because of an uauthorized scope. Adversaries may attempt to retrieve access tokens using client credentials to bypass user authentication and access resources. This query identifies when a public client app fails to retrieve an access token using client credentials and scopes that are not implicitly granted.
+"""
+integration = ["okta"]
+uuid = "0b936024-71d9-11ef-a9be-f661ea17fbcc"
+name = "Failed OAuth Access Token Retrieval via Public Client App with Client Credentials"
+language = ["ES|QL"]
+license = "Elastic License v2"
+notes = [
+    "Review `okta.debug_context.debug_data.flattened.grantType` to identify if the grant type is `client_credentials`",
+    "Ignore `okta.debug_context.debug_data.flattened.requestedScopes` values that indicate read-only access",
+    "Review `okta.actor.display_name` to identify the public client app that attempted to retrieve the access token. This may help identify the compromised client credentials.",
+    "Pivot for successful access token retrieval by the same public client app by searching `event.action` equal to `app.oauth2.as.token.grant.access_token` where the display name is the same."
+]
+mitre = ['T1550.001']
+query = ['''
+from logs-okta.system*
+| where @timestamp > NOW() - 7 day
+| where
+    event.dataset == "okta.system"
+
+    // filter on failed access token grant requests where source is a public client app
+    and event.action == "app.oauth2.as.token.grant"
+    and okta.actor.type == "PublicClientApp"
+    and okta.outcome.result == "FAILURE"
+
+    // filter out Elastic Okta integrations user agent
```

Comment:
risk is an adversary spoof the same user agent, maybe it can be filtered out using frequency/stats with high count high likely legit.

Reply:
Removed, the display name records the actual name of the client app so that should be enough